src/test/ui/consts/issue-73976-monomorphic.rs

```diff
@@ -5,20 +5,9 @@
 // will be properly rejected. This test will ensure that monomorphic use of these
 // would not be wrongly rejected in patterns.
 
-#![feature(const_type_id)]
 #![feature(const_type_name)]
 
-use std::any::{self, TypeId};
-
-pub struct GetTypeId<T>(T);
-
-impl<T: 'static> GetTypeId<T> {
-    pub const VALUE: TypeId = TypeId::of::<T>();
-}
-
-const fn check_type_id<T: 'static>() -> bool {
-    matches!(GetTypeId::<T>::VALUE, GetTypeId::<usize>::VALUE)
```

Comment:
I don't think we should lose the ability to compare type ids in CTFE. Comparison is almost the only thing you can do with a type id. It's possible to make `==` work here, but if that's tricky to do without potentially accidentally stabilizing it in CTFE then I think we should look at adding an unstable const function to compare them.

Reply:
Note that `==` still works in CTFE, it's just pattern matching that's no longer allowed.

Reply:
Do you know if we already have a test case that covers that @carbotaniuman?

Reply:
Hmm, maybe I'm missing something but it doesn't appear that `TypeId::of::<T>() == TypeId::of::<U>()` works in CTFE as of `nightly` today:

```
error[[E0277]](https://doc.rust-lang.org/nightly/error-index.html#E0277): can't compare `TypeId` with `_` in const contexts
 --> src/main.rs:6:23
  |
6 |     TypeId::of::<T>() == TypeId::of::<U>()
  |                       ^^ no implementation for `TypeId == _`
  |
  = help: the trait `~const PartialEq<_>` is not implemented for `TypeId`
note: the trait `PartialEq<_>` is implemented for `TypeId`, but that implementation is not `const`
 --> src/main.rs:6:23
  |
6 |     TypeId::of::<T>() == TypeId::of::<U>()
  |                       ^^
```

As far as I can tell that would still be the case after this PR, except that we'd also lose the ability to compare type ids by matching them.

Reply:
Yeah, it looks like I was mistaken here, and I'll likely reopen this PR with more substantial work behind it.